fix(enableDebugTools): create AngularTools at ng.tools

danielcrisp opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

`ng.probe` and `ng.coreTokens` are clobbered when you use `enableDebugTools` because the `AngularTools` class overwrites `context.ng` like this:

``````
```javascript
context.ng = new AngularTools(ref);
```
``````

Source: https://github.com/angular/angular/blob/master/modules/%40angular/platform-browser/src/browser/tools/tools.ts#L30

This means extensions like Augury no longer work because they use `ng.probe`.

More info: https://github.com/angular/angular/issues/12002

**What is the new behavior?**

`AngularTools` is now created at `context.ng.tools` so existing methods and properties on `ng.` are not affected.

This also allows `disableDebugTools` to remove the tools when required.

Unit tests and docs have been modified to incorporate this new location.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Fixes #12002

vicb commented on modules/@angular/platform-browser/src/browser/tools/common_tools.ts:
```diff
@@ -29,7 +29,7 @@ export class AngularTools {
 /**
  * Entry point for all Angular profiling-related debug tools. This object
- * corresponds to the `ng.profiler` in the dev console.
+ * corresponds to the `ng.tools.profiler` in the dev console.
```
is there a way to fix this without breaking the API ?

laco0416 replied:
can't AngularTools has `probe` field?

danielcrisp replied:
@vicb Yes, I could merge `AngularTools` into `context.ng` instead. Can I use `Object.assign`? I don't see it in use anywhere else in the source

danielcrisp replied:
I'm not keen on [this](https://github.com/angular/angular/issues/12002#issuecomment-250851200) suggestion because if additional methods are added to `ng.` someone would need to remember to keep this code up to date each time

danielcrisp committed:
fix(enableDebugTools): create AngularTools by merging into context.ng

When using `enableDebugTools` the tools are merged into `context.ng` to prevent `ng.probe` and `ng.coreTokens` from being clobbered.

Fixes #12002

danielcrisp commented:
Updated using a merge approach instead to maintain the `ng.profiler` API
